src/app/api/users/[userId]/notifications/route.ts

```diff
@@ -0,0 +1,33 @@
+import { getServerSession } from "next-auth";
+import { NextRequest, NextResponse } from "next/server";
+import { User } from "../../../../../lib/models/user.model";
+import dbConnect from "../../../../../lib/mongodb/mongodb";
+import { UserModel } from "../../../../../lib/schemas/user.schema";
+import authOptions from "../../../auth/auth-options";
+
+export async function GET(req: NextRequest, { params }: { params: { userId: string } }) {
+    const _id = params.userId;
+
+    const session = await getServerSession(authOptions);
+    if (!session || !session.user) {
+        return NextResponse.json({ error: "Unauthorized" }, { status: 401 });
+  }
+
+  await dbConnect();
+  try{
+      const user = await UserModel.findById(_id).select('notifications');
```

Comment:
You don't really need the select statement here since you're explicitly only returning the notifications (on Line 26)